xrpl/models/transactions/oracle_set.py

```diff
@@ -0,0 +1,128 @@
+"""Model for OracleSet transaction type."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from typing import Dict, List, Optional
+
+from xrpl.models.nested_model import NestedModel
+from xrpl.models.required import REQUIRED
+from xrpl.models.transactions.transaction import Transaction
+from xrpl.models.transactions.types import TransactionType
+from xrpl.models.utils import require_kwargs_on_init
+
+MAX_ORACLE_DATA_SERIES = 10
+MAX_ORACLE_PROVIDER = 256
+MAX_ORACLE_URI = 256
+MAX_ORACLE_SYMBOL_CLASS = 16
+
+
+@require_kwargs_on_init
+@dataclass(frozen=True)
+class OracleSet(Transaction):
+    """Represents a OracleSet transaction."""
+
+    account: str = REQUIRED  # type: ignore
+    """Account is the XRPL account that has update and delete privileges on the Oracle
```

Comment:
The docstrings should be copied from the documentation: https://opensource.ripple.com/docs/xls-47d-price-oracles/transaction-types/setoracle/

Reply:
Ditto for `OracleDelete`.

Reply:
fixed in [3957867](https://github.com/XRPLF/xrpl-py/pull/701/commits/3957867b12e9b76120b97bffd6a9d8d1b64f3cfa)